slide_analysis/UI/image_viewer.py

```diff
@@ -3,108 +3,190 @@
 # from PyQt5.QtWidgets import (QAction, QApplication, QFileDialog, QLabel,
 #                              QMainWindow, QMenu, QMessageBox, QScrollArea, QSizePolicy)
 #
+from PyQt5 import QtCore
+
 from PyQt5.QtCore import QDir, Qt
 from PyQt5.QtGui import QPalette, QPixmap
-from PyQt5.QtWidgets import QMainWindow, QFileDialog, QMessageBox, QAction, QApplication, QMenu
+from PyQt5.QtWidgets import *
 
 from slide_analysis.UI.image_helper import ImageHelper
 from slide_analysis.UI.ui_mainWindow import Ui_MainWindow
+from slide_analysis.UI.constants import SIMILAR_TILE_SIZE
+
+# class MyPopup(QWidget):
+#     def __init__(self, tile):
+#         QWidget.__init__(self)
+#         self.tile_label = QLabel(self)
+#         pixmap = QPixmap.fromImage(tile)
+#         self.tile_label.setPixmap(pixmap)
+#         self.resize(pixmap.width(), pixmap.height())
+#         position = self.cursor().pos()
+#         # position.setX(position.x() - thumb.size().width())
+#         # position.setY(position.y() - thumb.size().height())
+#         self.move(position)
+#         self.setWindowFlags(Qt.Popup | Qt.WindowStaysOnTopHint
+#                             | Qt.FramelessWindowHint
+#                             | Qt.X11BypassWindowManagerHint)
+#
+#     def keyPressEvent(self, event):
+#         if event.key() == QtCore.Qt.Key_Space:
+#             self.close()
+#             event.accept()
+#         else:
+#             super().keyPressEvent(event)
+
+
+class ImagePopup(QLabel):
```

Comment:
rename `ImagePopup` to name from which use cases can be determined like `TilePreviewPopupOnClick`
and move this class to separate module